libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1367,6 +1368,30 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
            }
         break;
     }
+    
+    case AUX_FUNC::MAG_CAL: {
+        Compass &compass = AP::compass();
+        switch (ch_flag) {
+        case AuxSwitchPos::LOW:
+            compass.cancel_calibration_all();
+            break;
+        case AuxSwitchPos::MIDDLE:
+            // nothing
+            break;
+        case AuxSwitchPos::HIGH:
+            if (!hal.util->get_soft_armed()) {
+                const bool retry = true;
+                const bool autosave = true;
+                const float delay = 5.0;
+                const bool autoreboot = true;
```

Comment:
Huh.  That's a brave stance.  I like it, I think.

Reply:
I'd like to get input from @rmackay9 on this .... do we really want to reboot?

Reply:
thinking was it would be another indicator that cal completed to those without GCS and only TX....Copter has this option, but was not the default...

Reply:
per our discussions, eliminated the reboot